cli/src/config.rs

```diff
@@ -1024,6 +1024,9 @@ pub struct _Validator {
     // Warp the ledger to WARP_SLOT after starting the validator.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub warp_slot: Option<String>,
+    // Whether to deploy the program as upgradeable. On real validator, this field is (program address, program.so, upgrade authority address).
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub upgradeable_program: Option<bool>,
```

Comment:
It would be better if we kept the same convention for the test validator arguments.

```suggestion
    pub upgradeable_program: Option<Vec<UpgradeableProgramEntry>>,
```

Reply:
After pushing up the `upgradeable_program` setting from `Validator` into `TestValidator` and `TestValidator.genesis`, we no longer need `UpgradeableProgramEntry` nor `BpfProgramEntry`